python/pyspark/sql/connect/ml/utils.py

```diff
@@ -0,0 +1,55 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pyspark.sql.utils import is_remote
+import os
+
+
+def _get_remote_ml_class(cls):
+    remote_module = "pyspark.sql.connect.ml." + cls.__module__[len("pyspark.ml."):]
+    cls_name = cls.__name__
+    m = __import__(remote_module, fromlist=[cls_name])
+    remote_cls = getattr(m, cls_name)
+    return remote_cls
+
+
+def try_remote_ml_class(x):
```

Comment:
I feel we can also simplify the `pyspark.sql` side by only using this annotation to the a few key classes

cc @HyukjinKwon